Lesson4

OlgaMyazina opened the pull request:
Запуск проекта командой `npm run dev`. Сервер работает на 8000 порту.

OlgaMyazina committed:
Установка express, первый успешный запуск на 8000 порту

OlgaMyazina committed:
Подключение nodemon

OlgaMyazina committed:
Разница во времени

OlgaMyazina committed:
Отдаёт json по api/events

OlgaMyazina committed:
Добавлена фильтрация по типу

OlgaMyazina committed:
Исправлена фильтрация событий

OlgaMyazina committed:
подключила cors

OlgaMyazina committed:
Работает fetch на получение events.json. Запуск одной командой npm run dev

Arinteck commented on server/data/config.json:
```diff
@@ -0,0 +1,4 @@
+{
+  "types": ["info", "critical"],
```
Про это мы с тобой проговорили: лучше фильтровать сразу по events.json, не нужно держать типы отдельно.

Arinteck commented on server/server.js:
```diff
@@ -0,0 +1,79 @@
+}
+
+/*
+app.use((err, request, response, next) => {
```
Удаляем

Arinteck commented on src/index.js:
```diff
@@ -1,6 +1,6 @@
 'use strict';
 
-const urlJSON = 'data/events.json';
+const urlJSON = 'http://localhost:8000/api/events';
```
👍

Arinteck commented on server/data/events.json:
```diff
@@ -0,0 +1,160 @@
+{
```
Сейчас два файла events.json, нужно оставить один

Arinteck commented on server/server.js:
```diff
@@ -0,0 +1,79 @@
+const data = require('./data/events');
```
Файлик нужно было считывать с помощью fs.readFile

Arinteck commented on package.json:
```diff
@@ -20,12 +20,15 @@
     "gh-pages": "^2.0.1",
     "handlebars-loader": "^1.7.0",
     "mini-css-extract-plugin": "^0.4.4",
+    "nodemon": "^1.18.4",
```
👍 Хорошо, что используешь nodemoon

Arinteck commented on server/server.js:
```diff
@@ -0,0 +1,79 @@
+  //Если не указан limit, то будет 10
+  //Если не указана page, то будет 1
+  if (request.query.page) {
+    const limit = request.query.limit ? parseInt(request.query.limit) : config.pageLimit;
```
У parseInt нужно обязательно указывать второй аргумент, могут быть ошибки: https://developer.mozilla.org/ru/docs/Web/JavaScript/Reference/Global_Objects/parseInt

Arinteck commented on server/server.js:
```diff
@@ -0,0 +1,79 @@
+  }
+  //Если не указан limit, то будет 10
+  //Если не указана page, то будет 1
+  if (request.query.page) {
```
Здорово, что есть пагинация!
Лучше было бы завязаться на request.query.limit, поскольку логично, что может быть указан только limit, но не указан page (хотим получить первых N-записей). А вот получить page без limit мы не можем, так как не знаем, сколько может быть на странице записей.

Arinteck commented on server/server.js:
```diff
@@ -0,0 +1,79 @@
+    const expectedTypes = config.types;
+    const queryTypes = request.query.type.split(':');
+    //проверяем типы из запроса queryTypes на наличие в конф. types
+    const hasIncorrectType = queryTypes.some(type => !expectedTypes.includes(type));
```
Проговорили с тобой, лучше сразу фильтровать по type событий, expectedTypes не нужен

Arinteck commented:
Все не плохо, все ручки работают корректно.
 
На будущее если заводишь pr с кодом, отбранчованым от какой-то ветки, у pr base branch должна быть эта ветка.

OlgaMyazina committed:
По замечанию: https://github.com/OlgaMyazina/ya-shri/pull/2#discussion_r225403778
Убрала типы из файла config.json, теперь типы собираются во множество из events.json

По замечанию: https://github.com/OlgaMyazina/ya-shri/pull/2#discussion_r225403924
Удаляем куски кода, которые находятся в комментарии

OlgaMyazina committed:
По замечанию: https://github.com/OlgaMyazina/ya-shri/pull/2#discussion_r225405268
Удалила файл events.json, который не использовался

OlgaMyazina committed:
Замечание: https://github.com/OlgaMyazina/ya-shri/pull/2#discussion_r225423753
замена способа получения events.json

OlgaMyazina committed:
Замечание: https://github.com/OlgaMyazina/ya-shri/pull/2#discussion_r225429647
второй аргумент у parseInt - указание системы счисления

OlgaMyazina committed:
Замечание: https://github.com/OlgaMyazina/ya-shri/pull/2#discussion_r225430166
Решила, что нужно ввести значение по-умолчанию и для limit и для page
